### PR TITLE
Clean incubator tests

### PR DESCRIPTION
**Pull request by nomisRev**
@rachelcarmena @danimontoya we're refactoring the test modules to be `arrow-optics-test`, arrow-fx-test`, etc with packages `arrow.optics.test`, `arrow.fx.test`, etc

Shall we for incubator already follow this format? I.e. should we move mtl laws into `arrow.mtl.test` package and laws for recursion in `arrow.recursion.test` package? If yes, then I will update this PR to reflect those changes :)

cc\\ @1Jajen1

### CODE REVIEW

**Comment by rachelcarmena** [comment]
> Shall we for incubator already follow this format? I.e. should we move mtl laws into `arrow.mtl.test` package and laws for recursion in `arrow.recursion.test` package? If yes, then I will update this PR to reflect those changes :)

I think it's a good idea :+1:

**Comment by 1Jajen1** [comment]
> Shall we for incubator already follow this format? I.e. should we move mtl laws into arrow.mtl.test package and laws for recursion in arrow.recursion.test package? If yes, then I will update this PR to reflect those changes :)

I guess so. mtl is likely the most used part of incubator, followed by free and then recursion schemes and they are probably not used together all that much so different packages make sense. Also helps if those things move out of incubator into their own repos which they should once they become more stable.

**Comment by 1Jajen1** [comment]
Btw the failing test is a correct test failure. @danimontoya recently changed foldable laws to properly test if `foldRight` is lazy in applying the function. In the case of `Free.Pure` it is not: Wrapping it in `defer` will work.

``` diff
 override fun <A, B> Kind<FreePartialOf<F>, A>.foldRight(lb: Eval<B>, f: (A, Eval<B>) -> Eval<B>): Eval<B> =
    fix().foldStep(
-     onPure = { a -> f(a, lb) },
+     onPure = { a -> defer { f(a, lb) } },
      onSuspend = { fa: Kind<F, A> -> FF().run { fa.foldRight(lb, f) } },
      onFlatMapped = { fa: Kind<F, A>, g: (A) -> Free<F, A> -> FF().run { fa.foldRight(lb) { a: A, lbb: Eval<B> -> g(a).foldRight(lbb, f) } } }
    )
```
In https://github.com/arrow-kt/arrow-incubator/blob/master/arrow-free/src/main/kotlin/arrow/free/extensions/free.kt#L105 should do it I think.

**Comment by rachelcarmena** [comment]
I'm downloading the branch to fix the build error :raised_hands:

File: arrow-mtl-data/src/test/kotlin/arrow/mtl/EitherTTest.kt [hunk 1]
```diff
@@ -35,15 +35,15 @@ import arrow.mtl.extensions.eithert.functor.functor
...
-import arrow.test.laws.ConcurrentLaws
-import arrow.test.laws.DivisibleLaws
-import arrow.test.laws.MonadErrorLaws
-import arrow.test.laws.TraverseLaws
-import arrow.test.laws.throwableEq
+import arrow.core.test.UnitSpec
+import arrow.core.test.generators.genK
+import arrow.core.test.generators.throwable
+import arrow.core.test.laws.AlternativeLaws
+import arrow.core.test.laws.ConcurrentLaws
```
Comment (rachelcarmena) [inline arrow-mtl-data/src/test/kotlin/arrow/mtl/EitherTTest.kt]:
```suggestion
import arrow.fx.test.laws.ConcurrentLaws
```

and it's necessary to add `arrow-fx-test` dependency.

**Comment by rachelcarmena** [comment]
@danimontoya , please, it's necessary to fix the data into: https://github.com/arrow-kt/arrow-incubator/blob/master/arrow-incubator-test/gradle.properties

File: arrow-incubator-test/src/main/kotlin/arrow/test/generators/GenK.kt [hunk 2]
```diff
@@ -1,130 +0,0 @@
...
-  }
-
-fun <A> Const.Companion.genK(genA: Gen<A>) = object : GenK<ConstPartialOf<A>> {
-  override fun <T> genK(gen: Gen<T>): Gen<Kind<ConstPartialOf<A>, T>> =
-    genA.map {
-      Const<A, T>(it)
-    }
-}
-
-fun <L, F> EitherT.Companion.genK(genkF: GenK<F>, genL: Gen<L>) =
```
Comment (danimontoya) [inline arrow-incubator-test/src/main/kotlin/arrow/test/generators/GenK.kt]: @aballano found it 😅

**Pull request merged**